#!/usr/bin/env python
PACKAGE='cargo_ants_slam'
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

#       Name                       Type       Reconfiguration levexl            Description                       Default   Min   Max
# general params
gen.add("verbose_mode", bool_t,  -1 , "verbose mode [t/f]", False)
gen.add("slam_rate", double_t, -1, "mapping rate [hz]", 10.0, 2.0, 20.0 )
gen.add("num_lasers", int_t,  -1 , "number of laser devices [#]",  1, 0, 10)
gen.add("base_link_name", str_t, 0, "Name of the vehicle base link", "agv_base_link")
gen.add("laser_idx_offset", int_t,  -1 , "Starting index to name laser frames [#]",  0, 0, 10)
gen.add("laser_link_name", str_t, 0, "Name of the laser device frames, without laser id", "agv_lidar")

# autogenerate class
exit(gen.generate(PACKAGE, "slam", "cargo_ants_slam_"))
